core/src/main/scala/chisel3/internal/firrtl/IR.scala

```diff
@@ -743,8 +743,10 @@ abstract class Definition extends Command {
 case class DefPrim[T <: Data](sourceInfo: SourceInfo, id: T, op: PrimOp, args: Arg*) extends Definition
 case class DefInvalid(sourceInfo: SourceInfo, arg: Arg) extends Command
 case class DefWire(sourceInfo: SourceInfo, id: Data) extends Definition
-case class DefReg(sourceInfo: SourceInfo, id: Data, clock: Arg) extends Definition
-case class DefRegInit(sourceInfo: SourceInfo, id: Data, clock: Arg, reset: Arg, init: Arg) extends Definition
+case class DefReg(sourceInfo: SourceInfo, id: Data, clock: Arg, reset: Option[Arg], init: Option[Arg]) extends Definition {
+  assert((reset.isDefined && init.isDefined) || (reset.isEmpty && init.isEmpty))
```

Comment:
what is the user experience if this fails? whats an example error message that they will get?

Reply:
User will never touch this code, since it's internal API. I'll add a error message if some-devs come into this.

Reply:
error message add in commit 919e965